### PR TITLE
Fix TypeError raised when using using default solver

### PR DESCRIPTION
**Pull request by jon-hagg**
### Purpose
If no solver is provided we should use the default which is determined in REISE.jl. I think I introduced this bug by adding the validation after having tested the code, and it was not re-tested (d'oh). 

### What the code is doing
Fix validation and add tests.

### Testing
New unit tests

### Time estimate
3 min

### CODE REVIEW

File: powersimdata/data_access/tests/test_launcher.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from powersimdata.data_access.launcher import _check_solver, _check_threads
+
+
+def test_check_solver():
+    _check_solver(None)
+    _check_solver("gurobi")
+    _check_solver("GLPK")
+    with pytest.raises(TypeError):
+        _check_solver(123)
+
```
Comment (BainanXia) [inline powersimdata/data_access/tests/test_launcher.py:12]: Do we need an empty line here?

Comment (jon-hagg) [reply powersimdata/data_access/tests/test_launcher.py]: Nope

**Review by BainanXia — APPROVED** [review]
I think this addresses the question brought up in our Slack community.

**Comment by jon-hagg** [comment]
> I think this addresses the question brought up in our Slack community.

Yeah I was going to respond but seems it was deleted? Can just message him directly though.

**Pull request merged**